src/api/spec/controllers/public_controller_spec.rb

```diff
@@ -1,14 +1,130 @@
     end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to eq(package.source_file("somefile.txt")) }
+  end
+
+  describe "GET #index" do
+    before do
+      get :index
+    end
+
+    it { is_expected.to respond_with(302) }
+    it { is_expected.to redirect_to(about_path) }
+  end
+
+  describe "GET #build" do
+    before do
+      get :build, project: project.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+  end
+
+
+  describe "GET #configuration" do
+    before do
+      get :configuration_show
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(assigns(:configuration)).to eq(Configuration.first) }
+  end
+
+  describe "GET #project_meta" do
+    before do
+      get :project_meta, project: project.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to match(/<project name="public_controller_project">/) }
+  end
+
+  describe "GET #project_index" do
+    before do
+      get :project_index, project: project.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to match(/<entry name="public_controller_package"/) }
+  end
+
+  describe "GET #project_file" do
+    before do
+      get :project_file, project: project.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+  end
+
+  describe "GET #project_file" do
+    context "_config" do
+      before do
+        get :project_file, project: project.name
+      end
+
+      it { is_expected.to respond_with(:success) }
+      it { expect(response.body).to eq(project.source_file("_config")) }
+    end
+  end
+
+  describe "GET #package_index" do
+    before do
+      get :package_index, project: project.name, package: package.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to match(/name="somefile.txt"/) }
+  end
+
+  describe "GET #package_meta" do
+    before do
+      get :package_meta, project: project.name, package: package.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to match(/<package name="public_controller_package" project="public_controller_project">/) }
+  end
+
+  describe "GET #distributions" do
+    before do
+      get :distributions, format: :xml
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(assigns(:distributions)).to eq(Distribution.all_as_hash) }
+  end
+
+  describe "GET #show_request" do
+    let(:request) { create(:bs_request) }
+
+    before do
+      get :show_request, number: request.number
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to eq(request.render_xml)}
+  end
+
+  describe "GET #binary_packages" do
+    before do
+      get :binary_packages, project: project.name, package: package.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(assigns(:pkg)).to eq(package) }
```

Comment:
Nope! I basically just covered the :success response and for some actions obvious / easy tests. Point is that this PR is about making /public route work for :html requests so my intention was not to add a complete test suite for the /public controller!